src/api/Tron.js

```diff
@@ -490,11 +491,27 @@ export const extractBandwidthInfo = (
   };
 };
 
-export const getTronResources = async (
+// This eslint rule has to be disabled for a known bug
+// @see https://github.com/gajus/eslint-plugin-flowtype/issues/374
+
+// eslint-disable-next-line no-redeclare
+declare function getTronResources(): Promise<TronResources>;
+// eslint-disable-next-line no-redeclare
+declare function getTronResources(
   acc: Object,
   txs: TrongridTxInfo[],
   cacheTransactionInfoById: { [_: string]: TronTransactionInfo }
-): Promise<TronResources> => {
+): Promise<TronResources>;
+// eslint-disable-next-line no-redeclare
+export async function getTronResources(
+  acc?: Object,
+  txs?: TrongridTxInfo[],
+  cacheTransactionInfoById?: { [_: string]: TronTransactionInfo }
+): Promise<TronResources> {
+  if (!acc) {
+    return defaultTronResources;
+  }
+
```

Comment:
Ok so best solution I got to allow for getTronResources to be called without params or with all params has been to overload the function and give it 3 declarations (1 without params, 1 with all the params, and 1 that is the combination of all which is mandatory). 

It's exactly what we want I guess, but unfortunately, between the ESLint plugin for flowtype not handeling the `declare` keyword correctly to ignore redeclaration + Flow just below that magically forget that the function is `async` and thinking that our return isn't a Promise, it's not most beautiful code in the world... 

I would need the opinion of other people on this. @haammar-ledger suggested to just let `getTronResources` as it was and just create another util to do a `getDefaultTronResources` which would work obviously but kinda lose the point of reuniting all the concerns about _getting tron resources_. 

I some of you have the time to give me an opinion on this, it would be greatly appreciated 😄

Reply:
Clarity has been resolved with the TS migration 👍